Plugins Docs (#295)
lightly-ai/lightly-studio: lightly_studio/docs/docs/index.md

### API Reference

See the [Python API](api/dataset.md) for more details on the python interface.

## Plugins

LightlyStudio offers the possibility to extend its functionality by using plugins. Users can define their own plugins or use available ones.

LightlyStudio plugins pair backend logic (the operator) with an optional frontend component. After you register an operator through the Python API, the GUI lists it automatically. For standard operators, the UI generates the input dialog for you.

In the following, we will see how operators are defined, and how we can use them to define a new plugin.

### Operator

> The core unit of a plugin is an operator. To create one you define some metadata, list the input parameters, and implement an execute method. Every operator follows the [`BaseOperator`](api/plugin/#lightly_studio.plugins.base_operator.BaseOperator) schema:

Reviewer: I would:
- make the docs such that the operator plugin is one type of plugin
- remove the repetition of what needs to be defined.
```suggestion
An operator plugin is defined by the following attributes of the [`BaseOperator`](api/plugin/#lightly_studio.plugins.base_operator.BaseOperator) schema:
```

        if label is None:
            label_create = AnnotationLabelCreate(annotation_label_name=name)
            label = annotation_label_resolver.create(session=session, label=label_create)

        label_map[name] = label.annotation_label_id

    return label_map

@dataclass
class LightlyTrainAutoLabelingODOperator(BaseOperator):
    name: str = "LightlyTrain: OD auto-labeling"
    description: str = "This plugin allows to use pre-trained LightlyTrain models to perform auto-labeling for Object Detection."

    @property
    def parameters(self):
        return [
            StringParameter(
                name="Model",
                required=True,
                description="The name of the pre-trained model to be used.",
                default="dinov3/convnext-tiny-ltdetr-coco"
            ),
            FloatParameter(
                name="Threshold",
                default=0.4,
                description="The confidence threshold to be applied to the predictions."
            ),
        ]

    def execute(self, *, session, dataset_id, parameters):
        try:
            model = lightly_train.load_model(parameters["Model"])
        except ValueError:

>             return OperatorResult(success=False, message="model_name is invalid.")

Reviewer: This is a risky, as there might be other `ValueError`s that are caught and then the error message is misleading. Best rather copy the error message from e of  `except ValueError as e` .

https://github.com/lightly-ai/lightly-train/blob/bb247bfc9989648a070e3a57dd9186d2018252a2/src/lightly_train/_task_models/task_model_helpers.py#L229
```suggestion
        try:
            model = lightly_train.load_model(parameters["Model"])
        except ValueError as e:
            return OperatorResult(success=False, message=f"Model load failed: {str(e)})
```
